app/mailers/patient_mailer.rb

```diff
@@ -20,60 +21,35 @@ def enrollment_email(patient)
   end
 
   def enrollment_sms_weblink(patient)
-    return if patient&.primary_telephone.blank?
+    # Should not be sending enrollment sms if no valid number or this patient is not a responder
+    return if patient&.primary_telephone.blank? || patient.id != patient.responder_id
 
     lang = patient.select_language
     patient_name = "#{patient&.first_name&.first || ''}#{patient&.last_name&.first || ''}-#{patient&.calc_current_age || '0'}"
     intro_contents = "#{I18n.t('assessments.sms.weblink.intro1', locale: lang)} #{patient_name} #{I18n.t('assessments.sms.weblink.intro2', locale: lang)}"
     url_contents = new_patient_assessment_jurisdiction_report_lang_url(patient.submission_token,
                                                                        lang&.to_s || 'en',
                                                                        patient.jurisdiction.unique_identifier[0, 32])
-    account_sid = ENV['TWILLIO_API_ACCOUNT']
-    auth_token = ENV['TWILLIO_API_KEY']
-    from = ENV['TWILLIO_SENDING_NUMBER']
-    client = Twilio::REST::Client.new(account_sid, auth_token)
-    client.messages.create(
-      from: from,
-      to: Phonelib.parse(patient.primary_telephone, 'US').full_e164,
-      body: intro_contents
-    )
-    client.messages.create(
-      from: from,
-      to: Phonelib.parse(patient.primary_telephone, 'US').full_e164,
-      body: url_contents
-    )
-  rescue Twilio::REST::RestError => e
-    Rails.logger.warn e.error_message
-    add_fail_history_sms(patient)
-    patient.update(last_assessment_reminder_sent: DateTime.now)
+    success = TwilioSender.send_sms(patient, intro_contents)
+    success &&= TwilioSender.send_sms(patient, url_contents)
```

Comment:
I think this line and the one below could just be a one liner if then else statement.